### DISCUSSION TITLE
How to convert query(exists()...) to SQLAlchemy 2 syntax?

### DISCUSSION THREAD

**Discussion by mcarans** [sqlalchemy/sqlalchemy#9404] [Usage Questions] [answered]
I have been trying to convert the following to SQLAlchemy 2 syntax:
```
                                if self.session.query(
                                    exists().where(
                                        and_(
                                            DBResource.id == resource_id,
                                            DBResource.md5_hash == hash,
                                        )
                                    )
                                ).scalar():
```

If I change `query` to `execute`, I get an error "Executable SQL or text() construct expected, got <sqlalchemy.sql.selectable.Exists object...". 

I'm not sure what I need to do to transform this to SQLAlchemy 2 syntax. Please advise.

**Comment by zzzeek** [comment, marked as answer]
you want to put the exists() inside of a select()

     stmt = select(exists().where(and_(...)))

run that

    result = session.scalar(stmt)

**Reply by aalvrz** [reply]
@zzzeek How to do this with multiple joins on different tables?

**Reply by CaselIT** [reply]
It's not really different that just generating a select. It should have `join` method.

**Reply by lokhman** [reply]
@aalvrz You wrap your select around exists:
```python
stmt = select(select(1).select_from(...).join(...).where(...).exists())
result = session.scalar(stmt)
```
Also equivalent to:
```python
stmt = select(exists(select(1).select_from(...).join(...).where(...)))
result = session.scalar(stmt)
```
